### PR TITLE
(ui) Update any-sequencer dependency parsing

### PR DESCRIPTION
**Pull request by drewbutlerbb4**
**Which issue is resolved by this Pull Request:** 
Resolves #

**Description of your changes:**
Parsing dependencies' for any-sequencers used to be based on metadata annotations. Those metadata annotations have been removed and now dependencies can be parsed from arguments in the taskSpec. This PR addresses this and collects the dependencies from the arguments in the taskSpec. 

This PR also allows conditional dependencies to be linked as well.

**Environment tested:**

* Python Version (use `python --version`):
* Tekton Version (use `tkn version`):
* Kubernetes Version (use `kubectl version`):
* OS (e.g. from `/etc/os-release`):

**Checklist:**
- [ ] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 

- [ ] Do you want this pull request (PR) cherry-picked into the current release branch?

    [Learn more about cherry-picking updates into the release branch](https://github.com/kubeflow/pipelines/blob/master/RELEASE.md#cherry-picking-pull-requests-to-release-branch).

### CODE REVIEW

**Comment by Tomcli** [comment]
thanks @drewbutlerbb4, the static graph is good. However, when you try to run this any-sequencer example, the dynamic graph has error until task1 is completed. 
https://github.com/kubeflow/kfp-tekton/blob/master/sdk/python/tests/compiler/testdata/any_sequencer.yaml

**Comment by Tomcli** [comment]
```
Utils.tsx:32 Error loading run: 59784800-def0-4d46-a399-7701008d583e TypeError: Cannot read property 'status' of undefined
    at WorkflowParser.ts:162
    at Array.forEach (<anonymous>)
    at WorkflowParser.ts:146
    at Function.value (WorkflowParser.ts:108)
    at n.<anonymous> (RunDetails.tsx:744)
    at u (runtime.js:63)
    at Generator._invoke (runtime.js:293)
    at Generator.next (runtime.js:118)
    at r (asyncToGenerator.js:3)
    at s (asyncToGenerator.js:25)
```

File: frontend/src/lib/WorkflowParser.ts [hunk 1]
```diff
@@ -141,6 +131,42 @@ export default class WorkflowParser {
...
+                const parentId =
+                  statusMap.get(parentTask)!['status']['podName'] ||
+                  statusMap.get(parentTask)!['pipelineTaskName'];
+                graph.setEdge(parentId, taskId);
+              });
+              isNextTaskList = false;
+            } else if (isNextCondition) {
+              const parentTask = arg.substring(arg.indexOf('results_') + 8, arg.indexOf('_output'));
+              const parentId =
+                statusMap.get(parentTask)!['status']['podName'] ||
```
Comment (Tomcli) [inline frontend/src/lib/WorkflowParser.ts:162]: probably have some fall back if the condition pod is not running yet.

**Comment by drewbutlerbb4** [comment]
Thanks @Tomcli should be fixed now.

**Comment by Tomcli** [comment]
thanks @drewbutlerbb4 
/lgtm

**Pull request merged**